File: Data Structures & Algorithms/reverse-linked-list-ii/submission-0.cpp
// struct ListNode
// {
//     int val;
//     ListNode* next;

//     ListNode() : val(0), next(nullptr), prev(nullptr) {}
//     ListNode(int val) : val(val), next(nullptr), prev(nullptr) {}
//     ListNode(int val, ListNode* next, ListNode* prev)
//         : val(val), next(next), prev(prev) {}
// };

class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        if (!head || left == right) return head;

        ListNode* cur = head;
        int idx = 1;

        stack<int> st;

        while (idx < left) {
            cur = cur->next;
            idx++;
        }

        ListNode* temp = cur;
        while (idx >= left && idx <= right) {
            st.push(temp->val);
            temp = temp->next;
            idx++;
        }

        while (!st.empty()) {
            cur->val = st.top();
            st.pop();
            cur = cur->next;
        }

        return head;
    }
};

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        Stack<Integer> stack = new Stack<>(); ListNode ptr = head; int size = 0; while (ptr != null) { size++; ptr = ptr.next; } ptr = head; for (int i = 0; i < size / 2; i++) { stack.push(ptr.val); ptr = ptr.next; } if (size % 2 == 1) { ptr = ptr.next; } while (ptr != null) { if (stack.pop() != ptr.val) { return false; } ptr = ptr.next; } return true;
    }
}